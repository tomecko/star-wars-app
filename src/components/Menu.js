import React from 'react';
import { Button, ButtonGroup, Card, CardBody, CardTitle } from 'reactstrap';

import { RESOURCE_INFOS } from '../config';

export function Menu({ resourceInfo, send }) {
  return (
    <Card>
      <CardBody>
        <CardTitle>Please choose battle type.</CardTitle>
        <ButtonGroup>
          {RESOURCE_INFOS.map(info => (
            <Button
              key={info.name}
              color="primary"
              onClick={() => {
                send({
                  type: 'PLAY',
                  resourceInfo: info,
                });
              }}
            >
              {info.name}
              {' '}
              {resourceInfo === info ? <small>(continue)</small> : null}
            </Button>
          ))}
        </ButtonGroup>
      </CardBody>
    </Card>
  );
}

