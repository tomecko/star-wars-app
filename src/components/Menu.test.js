import React from 'react';
import renderer from 'react-test-renderer';

import { Menu } from './Menu';

test('matches snapshot', () => {
  const component = renderer.create(
    <Menu resourceInfo={null} send={() => {}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
