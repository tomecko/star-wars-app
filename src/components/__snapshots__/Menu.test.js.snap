// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title"
    >
      Please choose battle type.
    </div>
    <div
      className="btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        people
         
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        starships
         
      </button>
    </div>
  </div>
</div>
`;
